packages/compose/src/hooks/use-event/index.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useInsertionEffect, useCallback } from '@wordpress/element';
+
+/**
+ * Any function.
+ */
+export type AnyFunction = ( ...args: any ) => any;
+
+/**
+ * Creates a stable callback function that has access to the latest state and
+ * can be used within event handlers and effect callbacks. Throws when used in
+ * the render phase.
+ *
+ * @param callback The callback function to wrap.
+ *
+ * @example
+ *
+ * ```tsx
+ * function Component(props) {
+ *   const onClick = useEvent(props.onClick);
+ *   React.useEffect(() => {}, [onClick]);
+ * }
+ * ```
```

Comment:
Should we use the [RFC example](https://github.com/reactjs/rfcs/blob/useevent/text/0000-useevent.md#basic-example)? Makes it clearer how `useEvent` is useful IMHO.

Reply:
I think in any case we should have at least one example of this being used in an effect, since the point is that it is untracked. So maybe we can add two examples: one in an effect, and one as an event handler.

Reply:
Two examples sound good 👍

Reply:
Looks great, thanks! Left minor styling suggestions in another comment.